components/library/gd2/Makefile

```diff
@@ -79,23 +100,86 @@ CONFIGURE_OPTIONS +=	--disable-rpath
 CONFIGURE_OPTIONS +=	--with-x
 CONFIGURE_OPTIONS +=	--with-pic
 
+# Unique stuff for the newer version
+BUILD_DIR_32_1 = $(COMPONENT_DIR)/build/$(COMPONENT_VERSION_1)-$(MACH32)
+BUILD_DIR_64_1 = $(COMPONENT_DIR)/build/$(COMPONENT_VERSION_1)-$(MACH64)
+
+# Override the unsuffixed vars to avoid "build/amd64" etc.
+BUILD_DIR_32 =	$(BUILD_DIR_32_1)
+BUILD_DIR_64 =	$(BUILD_DIR_64_1)
+
+# Note: linting rules are not quite fit for multi-source components like this
+# so we work around it with much complexity below... and still only get one
+# set of *.ln files built so far. Also they may be useless in practice and
+# just get dropped in a later round of recipying.
+include $(WS_MAKE_RULES)/lint-libraries.mk
```

Comment:
I'd rather just removed lint libraries, as gd is not used by illumos-gate build and removed additional complexity.